### DISCUSSION TITLE
scene change dosnt happen on web

### DISCUSSION THREAD

**Discussion by meoowe** [korlibs/korge#2033] [Q&A] [answered]
I am quite new to KorGE and am making a game for a game jam. On my title screen i have a uiButton labled `Play`. If i run my game using `runJs` and click the button, nothing happens. It should change to a different scene but instead it stays on the title screen.

my play button code
`
val playButton = uiButton("Play") {
            position(350, 200)
        }
`
my onClick listener code
`
playButton.onClick {
            sceneContainer.changeTo{ GameScene() }
            sceneDestroy()
 }`

**Comment by maik-mursall** [comment]
@meoowe Heya :)
Does this also happen, when you remove the `sceneDestroy()` call?

**Reply by meoowe** [reply]
yes it still happens :(

**Reply by maik-mursall** [reply]
Sorry for the late reply 😅 
Which Korge Version are you using?

Using the "Hello World"-Template (Korge 5.0.6) the following works for me without any problems:

```kotlin
suspend fun main() = Korge(windowSize = Size(512, 512), backgroundColor = Colors["#2b2b2b"]) {
    val sceneContainer = sceneContainer()

    sceneContainer.changeTo { TitleScreen() }
}

class TitleScreen : Scene() {
    override suspend fun SContainer.sceneMain() {
        val playButton = uiButton("Play") { position(350, 200) }
        playButton.onClick {
            sceneContainer.changeTo { GameScreen() }
        }
    }
}

class GameScreen : Scene() {
    override suspend fun SContainer.sceneMain() {
        text("Game Scene")
            .centerOnStage()
    }
}
```

Bumping the version to 5.1.0 works as well without any problems :)

**Reply by meoowe** [reply]
Im using Korge 5.0.6 and have not updated since creating my project. My code is the same as yours exept the GameScreen screen is in a different file.

**Reply by maik-mursall** [reply]
Does this even happen, if you copy and run my code within your project? :)

Do you maybe get any errors in the console while navigating?
Maybe you have an Error within your `sceneMain` or `sceneInit`.

**Comment by meoowe** [comment]
if i create a new project with your code it works perfectly fine. I tried commenting out everything and now I think the problem is this.
`val sound = resourcesVfs["Track.mp3"].readMusic()`
`sound.play(infinitePlaybackTimes)`

**Reply by maik-mursall** [reply, marked as answer]
I noticed, that with `readMusic` there is an console error being thrown on JS for me 🤔 
Does it work for you when you use `readSound` instead (You might need to wait a few seconds for the sound to start)

[Here](https://docs.korge.org/audio/#loading-sounds-and-music-files) is an explanation on the differences.

**Reply by maik-mursall** [reply]
In any case `readMusic` throwing an error and `readSound` doesn't is a bug for me at least 😅

**Reply by meoowe** [reply]
oh i thought readMusic would be better because it was quite a long track

**Reply by meoowe** [reply]
it works now! THANK YOU!!! For some reason i didnt get the error?

**Comment by maik-mursall** [comment]
> oh i thought readMusic would be better because it was quite a long track
Normally that would be better, since with `readSound` it loads the entire track into memory and `readMusic` streams it.

But for now I fear that seems to be your only option 😅 

Glad to help :D
